### PR TITLE
Disable view CRLP for non-Admin users

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
We eliminated a security issue which allowed non-Admin users to view, but not edit, Customized Rollup Settings.

# Critical Changes
 - We eliminated a security issue which allowed unauthorized users to view, but not edit, Customized Rollup Settings. User profiles must now have Customize Application, Modify All Data, and Author Apex system permissions to access Customized Rollup Settings.

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
**lurch: attach W-041077

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HvTLQA0" target="_blank">W-041077</a>

**Comment by npsp-reedestockton** [comment]
@force2b This PR is ready for review. Thanks!
@sam-knox This PR is ready for your review. No new labels, but please verify that stgEditPermissionError is appropriate for this usage.  Thanks!

File: src/aura/CRLP_RollupsContainer/CRLP_RollupsContainer.cmp [hunk 1]
```diff
@@ -56,6 +56,7 @@
     <aura:attribute name="isRollupDetail" type="Boolean" access="private" description="Determines if Rollups Detail is shown."/>
     <aura:attribute name="isFilterGroupDetail" type="Boolean" access="private" description="Determines if Filter Group Detail is shown."/>
     <aura:attribute name="isCRLPEnabled" type="Boolean" access="private" description="Determines if CRLP is enabled." default="true"/>
+    <aura:attribute name="isAdmin" type="Boolean" access="private" description="Determines if user is System Admin." default="true"/>
```
Comment (force2b) [inline src/aura/CRLP_RollupsContainer/CRLP_RollupsContainer.cmp:59]:
```suggestion
    <aura:attribute name="isSystemAdminUser" type="Boolean" access="private" description="Determines if user has the correct administrator permissions" default="true"/>
```

File: src/classes/CRLP_RollupUI_SVC.cls [hunk 2]
```diff
@@ -101,6 +101,7 @@ public with sharing class CRLP_RollupUI_SVC {
         @AuraEnabled public Map<String, String> labels;
         @AuraEnabled public List<FilterGroupTableItem> filterGroups;
         @AuraEnabled public Boolean isCRLPEnabled;
+        @AuraEnabled public Boolean isAdmin;
```
Comment (force2b) [inline src/classes/CRLP_RollupUI_SVC.cls:104]:
```suggestion
        @AuraEnabled public Boolean isSystemAdminUser;
```

File: src/classes/CRLP_RollupUI_SVC.cls [hunk 3]
```diff
@@ -147,9 +148,10 @@ public with sharing class CRLP_RollupUI_SVC {
 
         model.labels = getAppLabels();
         model.isCRLPEnabled = Customizable_Rollup_Settings__c.getOrgDefaults().Customizable_Rollups_Enabled__c;
+        model.isAdmin = STG_Panel.runningUserIsAdmin();
 
         //return early with just labels and CRLP enabled flag if CRLP is disabled
-        if (!model.isCRLPEnabled) {
+        if (!(model.isCRLPEnabled)) {
```
Comment (force2b) [inline src/classes/CRLP_RollupUI_SVC.cls:154]: Out of curiosity, why the extra set of parenthesis?

Comment (npsp-reedestockton) [reply src/classes/CRLP_RollupUI_SVC.cls:154]: I was previously testing both isCRLPEnabled and isSystemAdminUser, so the second set was previously needed.  Just forgot to remove the extra set.

**Comment by npsp-reedestockton** [comment]
@sam-knox We're checking for customize application, modify all data, and author apex permissions rather than the profile name. Customer could have customized profile names, so we need to check actual permissions.

**Comment by npsp-reedestockton** [comment]
@force2b PR change requests have been made. Ready for a 2nd look. Thanks!

**Comment by rponti-sforg** [comment]
QE testing for this PR and associated PR is now complete.  Based on the WI Description, Bug Bounty notes and comments from GUS, ACs, and Test Charter feedback, tests have passed.

For further testing details, see: 
https://foundation.lightning.force.com/lightning/r/agf__ADM_Work__c/a2x1E000001HvTLQA0/view
https://salesforce.quip.com/FcjhAuRytKqJ

Proceeding with merge of PR.  Once post-merge builds have passed, WI will be set to 'Closed.

**Pull request merged**